File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    var answer = 0;
    
    for(var i=0; i<s.length; i++){
        var otherCnt = 0;
        var xCnt = 0;
        var x = s[i];
        while(i<s.length){
            var ch = s[i++];
            if(ch !== x){
                otherCnt++;
            }
            else {
                xCnt++;
            }
            if(x !== ch && xCnt === otherCnt) break;
        }
        i--;
        answer++;
    }
    return answer;
}

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        int xcnt = 0;
        int ycnt = 0;
        char[] str = s.toCharArray();
        char x = str[0];
        for(int i=0; i<str.length; i++){
            if(xcnt==0){
                x = str[i];
                xcnt++;
                continue;
            }else if(str[i]==x){
                xcnt++;
            }else{
                ycnt++;
            }
            if(xcnt == ycnt){
                answer++;
                xcnt = 0;
                ycnt = 0;
            }
        }
        if(xcnt!=0)
            answer++;
        
        return answer;
    }
}